examples/java8/src/main/java/org/apache/beam/examples/complete/game/AggregatedTeamScore.java

```diff
@@ -0,0 +1,183 @@
+    tableConfigure.put(
+        "processing_time",
+        new WriteWindowedToBigQuery.FieldInfo<KV<String, Integer>>(
+            "STRING", (c, w) -> fmt.print(Instant.now())));
+    return tableConfigure;
+  }
+
+
+  public static void main(String[] args) throws Exception {
+
+    Options options = PipelineOptionsFactory.fromArgs(args).withValidation().as(Options.class);
+    // Enforce that this pipeline is always run in streaming mode.
+    options.setStreaming(true);
+    ExampleUtils exampleUtils = new ExampleUtils(options);
+    Pipeline pipeline = Pipeline.create(options);
+
+    pipeline
+        // Read game events from Pub/Sub using custom timestamps, which are extracted from the
+        // pubsub data elements, and parse the data.
+        .apply(PubsubIO.readStrings()
+            .withTimestampAttribute(TIMESTAMP_ATTRIBUTE).fromTopic(options.getTopic()))
+        .apply("ParseGameEvent", ParDo.of(new ParseEventFn()))
+        // Create <team, GameActionInfo> mapping. UpdateTeamScore uses team name as key.
+        .apply("MapTeamAsKey", MapElements
+            .into(TypeDescriptors.kvs(TypeDescriptors.strings(),
+                TypeDescriptor.of(GameActionInfo.class)))
+            .via((GameActionInfo gInfo) -> KV.of(gInfo.team, gInfo)))
+        // Outputs team score every time a team reaches the threshold.
+        .apply("UpdateTeamScore",
+            ParDo.of(new UpdateTeamScoreFn(options.getThreshold())))
+        // Write the results to BigQuery.
+        .apply(
+            "WriteTeamLeaders",
+            new WriteWindowedToBigQuery<KV<String, Integer>>(
+                options.as(GcpOptions.class).getProject(),
+                options.getDataset(),
+                options.getLeaderBoardTableName() + "_team_leader",
+                configureCompleteWindowedTableWrite()));
+
+    // Run the pipeline and wait for the pipeline to finish; capture cancellation requests from the
+    // command line.
+    PipelineResult result = pipeline.run();
+    exampleUtils.waitToFinish(result);
+  }
+
+  // Outputs team score every time a team reaches the threshold.
+  @VisibleForTesting
+  static class UpdateTeamScoreFn
+      extends DoFn<KV<String, GameActionInfo>, KV<String, Integer>> {
+
+    private final int threshold;
+
+    public UpdateTeamScoreFn(int threshold) {
+      this.threshold = threshold;
+    }
+
+    @StateId("totalScore")
+    private final StateSpec<ValueState<Integer>> totalScoreSpec =
+        StateSpecs.value(VarIntCoder.of());
+
+    @ProcessElement
+    public void processElement(
+        ProcessContext c,
+        @StateId("totalScore") ValueState<Integer> totalScore) {
+      String teamName = c.element().getKey();
+      GameActionInfo gInfo = c.element().getValue();
+
+      int oldTotalScore = firstNonNull(totalScore.read(), 0);
+      totalScore.write(oldTotalScore + gInfo.score);
+
+      if (oldTotalScore / this.threshold != totalScore.read() / this.threshold) {
```

Comment:
This is a very confusing `if` statement if you have a different idea of what "threshold" means.

There are two ways to think about how the pipeline runs, and my instinct was a different way than what is being done here - basically you output when the team has reached the threshold number of points _since the last time you output_ (e.g. threshold == 10; (+5, +5, +5, +5) gives you 10 and 20; (+7; +10; +5; +15) gives you 17 and 37; this pipeline outputs 17, 22, and 37)

Update comments and naming to reflect specifically what happens (every time the team passes a new multiple of the threshold is roughly how I would word it)

Reply:
added bunch of comments everywhere.